### PR TITLE
Endpoint for APIv4: GET /team/{team_id}/channels

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Implementation endpoint for APIv4: GET /team/{team_id}/channels

#### Ticket Link
Jira ticket: [PLT-5245](https://mattermost.atlassian.net/browse/PLT-5245)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Added API documentation (required for all new APIs) [mattermost-api-reference#132](https://github.com/mattermost/mattermost-api-reference/pull/132)
- [x] All new/modified APIs include changes to the drivers
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) updates
- [x] Touches critical sections of the codebase (channel store)

### CODE REVIEW

File: api4/channel.go [hunk 1]
```diff
@@ -115,6 +117,26 @@ func getChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPublicChannels(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (jwilander) [inline api4/channel.go]: Minor but can we rename this to `getPublicChannelsForTeam` so it's a bit more explicit? (Same with the client4.go and store functions)

File: store/sql_channel_store.go [hunk 2]
```diff
@@ -524,6 +524,40 @@ func (s SqlChannelStore) GetMoreChannels(teamId string, userId string, offset in
...
+
+		data := &model.ChannelList{}
+		_, err := s.GetReplica().Select(data,
+			`SELECT
+			    *
+			FROM
+			    Channels
+			WHERE
+			    TeamId = :TeamId
+					AND Type IN ('O')
```
Comment (jwilander) [inline store/sql_channel_store.go]: Can we do `Type = 'O'` instead? I know the database engine will probably optimize it anyway but it's nice to be more specific

**Comment by saturninoabril** [comment]
@jwilander noted. updated function names. thanks

File: config/config.json [hunk 3]
```diff
@@ -55,10 +55,10 @@
         "CustomDescriptionText": "",
         "RestrictDirectMessage": "any",
         "RestrictTeamInvite": "all",
-        "RestrictPublicChannelCreation": "all",
-        "RestrictPrivateChannelCreation": "all",
         "RestrictPublicChannelManagement": "all",
         "RestrictPrivateChannelManagement": "all",
+        "RestrictPublicChannelCreation": "all",
+        "RestrictPrivateChannelCreation": "all",
         "RestrictPublicChannelDeletion": "all",
```
Comment (saturninoabril) [inline config/config.json]: @jwilander i overlooked this ```config.json```. i'll revise the PR. thanks

Comment (jwilander) [inline config/config.json]: Seems I missed that as well, I through a WIP label on it for now

Comment (saturninoabril) [inline config/config.json]: done reset, then unstaged ```config.json```. thanks

**Comment by grundleborg** [comment]
Thanks for the PR @saturninoabril

**Pull request merged**